Feature profile backend

crispaulrubi opened the pull request:
Backend for profile changing. Wa pa na-integrate ang admin nga profile page since wa pa ang frontend

rosalierazonable commented:
## Findings
- Error when logged in as Admin
- Only works for student user type
- Conflict on variable names must be resolved, either set logic in the view for setting values depending on the user type or just use the same variable name.
- User's full name is hardcoded at the top of the page
- When clicking show password, it shows the "show PW here" other than the actual password. Should the password even be displayed? Is it right to display the (current) password in terms of security? 
- Change avatar doesn't work

## Clarifications
- What is the difference between SAVE and CHANGE action buttons? The save button modifies the database entry but the change does not do anything. It updates the database entry only after loading it a couple of times, please confirm the expected behavior
- The save and change button when clicked, displays a pop up that the data has been updated successfully even when I did not do any modifications. Is this the expected behavior?

crispaulrubi commented:
### Findings

> Error when logged in as Admin
> Only works for student user type
> Conflict on variable names must be resolved, either set logic in the view for setting values depending on the user type or just use the same variable name.
> 
This issue is because of the lack in frontend for admin profile. After I integrate it with Christian's admin profile, this issue will also be fixed.

> User's full name is hardcoded at the top of the page
FIXED

> When clicking show password, it shows the "show PW here" other than the actual password. Should the password even be displayed? Is it right to display the (current) password in terms of security?
This issue is to be removed by Christian. Password should not be shown. The reason is that we don't have any way to determine the password of the user since it is hashed, hence we could never display passwords. I told Christian to remove this input field and have "change" button only.

> Change avatar doesn't work
FIXED
### Clarifications

> What is the difference between SAVE and CHANGE action buttons? The save button modifies the database entry but the change does not do anything. It updates the database entry only after loading it a couple of times, please confirm the expected behavior.
> If there is no difference between save and change buttons shall we remove the other one?
> The save and change button when clicked, displays a pop up that the data has been updated successfully even when I did not do any modifications. Is this the expected behavior?
The "save" button will save the inputs to the database. The change button will redirect to Patrick's change password page. I changed the button to "Change Password" instead since it does lead to confusion.
